File: internal/domain/game.go
package domain

import "time"

type GameMode string
type GameStatus string

const (
	GameModeSolo GameMode = "solo"
	GameMode1v1  GameMode = "1v1"
)

const (
	GameStatusPlaying  GameStatus = "playing"
	GameStatusFinished GameStatus = "finished"
)

type Game struct {
	ID         int64      `json:"id"`
	MatchID    *int64     `json:"match_id,omitempty"`
	PlayerID   int64      `json:"player_id"`
	Mode       GameMode   `json:"mode"`
	Score      int64      `json:"score"`
	Status     GameStatus `json:"status"`
	StartedAt  time.Time  `json:"started_at"`
	FinishedAt *time.Time `json:"finished_at,omitempty"`
}

File: app/domain/game.py
from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class GameMode(str, Enum):
    SOLO = "solo"
    ONE_V_ONE = "1v1"


class GameStatus(str, Enum):
    PLAYING = "playing"
    FINISHED = "finished"


class Game(BaseModel):
    id: int | None = None
    match_id: int | None = None
    player_id: int
    mode: GameMode
    score: int = 0
    status: GameStatus = GameStatus.PLAYING
    started_at: datetime = datetime.now()
    finished_at: datetime | None = None
